server/lib/pbench/process_tb.py

```diff
@@ -15,20 +15,34 @@ def __init__(self, config: PbenchConfig, logger: Logger):
         and copies it to the remote server by creating a subprocess which
         call Agent's 'pbench-results-push' command.
 
-            Args -
-                config -- PbenchServer config object
-                logger -- logger object to use when emitting log entries during
-                            operation
+        Args:
+            config : PbenchServer config object
+            logger : logger object to use when emitting log entries during
+                operation
 
+        Raises:
+            ValueError when the pbench-receive-dir-prefix is not provided, or
+                when both a token and a set of dispatch-states are not provided
+            NotADirectoryError when the reception directory does not exist
         """
         self.config = config
         self.logger = logger
+        # The new pbench server PUT API token to use when copying tar balls to
+        # the remote server.
         self.token = self.config.get("pbench-server", "put-token")
-        if not self.token:
+        # Perform local directory-based dispatching for additional operations
+        # (e.g. unpacking tar balls, backups, etc.).
+        dispatch_states_str = self.config.get("pbench-server", "dispatch-states")
+        if not self.token and not dispatch_states_str:
             raise ValueError(
-                "No value for config option put-token in section pbench-server"
+                "The 'pbench-server' section must either provide a value for"
+                " the 'put-token' option, and/or the 'dispatch-states' option"
             )
```

Comment:
Interesting. So `put-shim` can be configured to operate in "prep-shim" mode, "put-shim" mode, or the new hybrid mode.